src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -765,7 +769,40 @@ class InvokeHostFunctionApplyHelper : virtual LedgerAccessHelper
         mOpFrame.innerResult(mRes).code(INVOKE_HOST_FUNCTION_SUCCESS);
         mOpFrame.innerResult(mRes).success() = xdrSha256(success);
 
-        mOpMeta.getEventManager().setEvents(std::move(success.events));
+        if (!mProtocol23SACReconciliationEvents.empty())
+        {
+            xdr::xvector<ContractEvent> events;
+            events.reserve(success.events.size() +
+                           mProtocol23SACReconciliationEvents.size());
+            for (auto const& rEvent : mProtocol23SACReconciliationEvents)
+            {
+                if (rEvent.amount > 0)
+                {
+                    events.emplace_back(mOpMeta.getEventManager().makeMintEvent(
+                        rEvent.asset, rEvent.mintOrBurnAddress, rEvent.amount,
+                        false));
+                }
+                else
+                {
+                    events.emplace_back(mOpMeta.getEventManager().makeBurnEvent(
+                        rEvent.asset, rEvent.mintOrBurnAddress,
+                        -rEvent.amount));
```

Comment:
`-rEvent.amount` is a bit non-obvious - maybe add a comment stating that we're in the 'negative amount' branch here?
Also, should we assert that amount is non-zero?